attestationreport/attestationreport.go

```diff
@@ -963,30 +969,33 @@ func verifyAndUnpackAttestationReport(attestationReport string, result *Verifica
 		}
 	}
 
-	// Validate and unpack Company Description
-	jwsValRes, payload, ok = VerifyJws(arJws.CompanyDescription, roots, r.CompanyDescSigners)
-	result.CompDescResult.Summary = jwsValRes.Summary
-	result.CompDescResult.SignatureCheck = jwsValRes.SignatureCheck
-	if !ok {
-		log.Trace("Verification of Company Description Signatures failed")
-		result.CompDescResult.Summary.Success = false
-		result.Success = false
-	}
-	err = json.Unmarshal(payload, &ar.CompanyDescription)
-	if err != nil {
-		msg := fmt.Sprintf("Unpacking of Company Description failed: %v", err)
-		result.CompDescResult.Summary.setFalseMulti(&msg)
-		result.Success = false
-	} else {
-		result.CompDescResult.Name = ar.CompanyDescription.DN
-		result.CompDescResult.CompCertLevel = ar.CompanyDescription.CertificationLevel
-
-		result.CompDescResult.ValidityCheck = checkValidity(ar.CompanyDescription.Validity)
-		if result.CompDescResult.ValidityCheck.Success == false {
-			log.Trace("Company Description invalid")
+	// Validate and unpack Company Description if present
+	if arJws.CompanyDescription != "" {
+		jwsValRes, payload, ok = VerifyJws(arJws.CompanyDescription, roots, r.CompanyDescSigners)
+		result.CompDescResult = &CompDescResult{}
+		result.CompDescResult.Summary = jwsValRes.Summary
+		result.CompDescResult.SignatureCheck = jwsValRes.SignatureCheck
+		if !ok {
+			log.Trace("Verification of Company Description Signatures failed")
 			result.CompDescResult.Summary.Success = false
```

Comment:
Summary.Success should ideally already be false since line 976 sets Summary as jwsValRes.Summary?